webapp/src/ts/components/search-bar/search-bar.component.ts

```diff
@@ -24,16 +35,24 @@ export class SearchBarComponent implements AfterContentInit, AfterViewInit, OnDe
   @Input() disabled;
   @Input() showFilter;
   @Input() showSort;
+  @Input() showBarcodeScanner;
   @Input() sortDirection;
   @Input() lastVisitedDateExtras;
   @Output() sort: EventEmitter<any> = new EventEmitter();
   @Output() toggleFilter: EventEmitter<any> = new EventEmitter();
   @Output() search: EventEmitter<any> = new EventEmitter();
 
+  private TELEMETRY_PREFIX = 'search_by_barcode';
+  private globalAction: GlobalActions;
+  private barcodeDetector;
   private filters;
+  private barcodeTypes;
+  private barcodeImageElement;
+  windowRef;
```

Comment:
I was about to suggest to type it as `Document['defaultView']` but it would not bring any benefit because typescript has no knowledge of `BarcodeDetector` and its APIs :disappointed:  
We're back to relying exclusively on the MDN docs

Reply:
yeah :( we won't be able to upgrade the TypeScript version for a while anyway because of Angular and the browser support.